ARCs/arc-draft_extended_desc.md

```diff
@@ -0,0 +1,160 @@
+---
+arc: <to be assigned>
+title: Extended App Description
+description: Adds more information to the ARC4 JSON description
+author: Joe Polny (@joe-p)
+discussions-to: <URL>
+status: Draft
+type: Standards Track
+category: ARC
+created: 2023-11-14
+requires: 4
+---
+
+
+## Abstract
+This ARC takes the existing JSON description of a contract as described in ARC4 and adds more fields for the purpose of client interaction
+
+## Motivation
+The description provided by ARC4 is missing a lot of critical information that clients should know when interaction with an app.
+
+## Specification
+```ts
+/** Mapping of named structs to the ABI type of their fields */
+interface StructFields {
+  [name: string]: string | StructFields;
+}
+
+/** Describes a single key in app storage */
+interface StorageKey {
+  /** The type of the key. Can be ABI type or named struct */
+  keyType: string;
+  /** The type of the value. Can be ABI type or named struct */
+  valueType: string;
```

Comment:
Wonder if this should be `StructName | ABIType` or similar?

Also, previously we had avm type, do we need to support that too / instead? Having the proper type would be awesome because it's an annoying limitation of the current stuff.

Reply:
> Wonder if this should be StructName | ABIType or similar?

Yeah I like that. Will make the change. 

> Also, previously we had avm type, do we need to support that too / instead? Having the proper type would be awesome because it's an annoying limitation of the current stuff.

Initially I was thinking it wasn't necessary because it's obvious when one is using uint64 vs ABI uint64, but we do need a way to specifcy AVM bytes (no prefix) vs ABI bytes. i'll add AVMType as well.

Reply:
I added `AVMBytes` type in a18a968a16f3bb272c5da8d8e8d90f18eddb7ab8. I don't think uint64 is needed because it's always going to be obvious when AVM uint64 is used (only top-level values in global/local)